package.json

```diff
@@ -22,8 +22,8 @@
     "lint": "yarn lint-solidity && yarn lint-js && yarn lint-rust",
     "lint-solidity": "yarn solhint ./contracts/**/*.sol",
     "lint-js": "yarn prettier --list-different **/*.js **/*.ts",
-    "lint-rust": "yarn prettier --list-different ./programs/**/*.rs",
-    "lint-fix": "yarn prettier --write **/*.js **/*.ts ./programs/**/*.rs ./contracts**/*.sol",
+    "lint-rust": "cargo fmt --all -- --check && cargo clippy",
```

Comment:
so this is the command that is run in [CI](https://github.com/across-protocol/contracts/blob/3cfd778c1bd6d805b10cbe91f794bb24b060c26a/.github/workflows/pr.yml#L44). as a result, we might want `lint-rust` to be:
```
yarn prettier --list-different ./programs/**/*.rs && cargo fmt --all -- --check && cargo clippy
```

i.e it mirrors what is done in `lint-fix` except errors if the linting is not applied correctly.

Reply:
I'm not sure on this. These are conflicting formats, so prettier would error if cargo fmt was run before. I think its more appropriate to use prettier only in pre-commit hook and `lint-fix` script before rustfmt, but we don't want to break CI if code does not confirm to prettier rules (as they might conflict with rustfmt)

Reply:
The pre-commit hook or `lint-fix` applies these in order. First, it runs prettier, but once it runs `rustfmt` this changes the formatting. As prettier is opinionated and differs from standard `rustfmt` settings then having `prettier` within `lint-rust` would error.

Reply:
I see. hhmm. that's a bit of a pain then that they've diverged like that. I guess if we want both prettier and rustfmt this is the path we need to take.

This means in some ways that the prettier formatter is optional but the rustfmt is compulsory (its the only linter enforced by CI). I think this is ok

Reply:
yes, we really need prettier to deal with anchor macros here